app/app.go

```diff
@@ -251,6 +263,25 @@ func (d *Deployments) handleArtifact(ctx context.Context,
 		return artifactID, uploadResponseErr
 	}
 
+	// Check that the submitted size is correct
+	var b [1]byte
+	N, err := multipartUploadMsg.ArtifactReader.Read(b[:])
+	if N > 0 && err == nil {
```

Comment:
I do not get this if, how is that checking the correctness of the size?

Reply:
It's a bit cheeky, but the `LimitReader` returns `io.EOF` after the given number of bytes are read. Notice that this is the underlying reader the `LimitReader` is reading from. Thus, if we read more bytes from the reader, the actual payload is longer than the given size.